lua/nvim-surround/init.lua

```diff
@@ -195,6 +193,34 @@ M.change_surround = function(args)
     local selections = utils.get_nearest_selections(args.del_char, "change")
     local delimiters = args.add_delimiters()
     if selections and delimiters then
+        -- Avoid adding any, and remove any existing whitespace after the
+        -- opening delimiter if only whitespace exists between it and the end
+        -- of the line. Avoid adding or removing leading whitespace before the
+        -- closing delimiter if only whitespace exists between it and the
+        -- beginning of the line.
+
+        -- Column number of end of opening selection
+        local xpos = selections.left.last_pos[2]
```

Comment:
Also I would either substitute the right hand side directly where you refer to `xpos` on line 206, or use a better name for `xpos`. It doesn't really mean anything to me, and you're using a comment to document the variable name anyways.

Reply:
You're right. It doesn't really explain anything and it doesn't save any space either. It probably made more sense while I was drafting the code, but by the time I was done I didn't notice it only got used in one place per assignment.

`xpos` to me carries a fairly sensible meaning of "horizontal position", but of course that's still pretty ambiguous, especially considering the only other two variables in each context *also* refer to "horizontal position", although their names are much more specific.